query/src/sessions/sessions.rs

```diff
@@ -115,8 +96,11 @@ impl SessionManager {
         self.catalog.clone()
     }
 
-    pub fn get_storage_cache(self: &Arc<Self>) -> Arc<Option<Box<dyn StorageCache>>> {
-        self.storage_cache.clone()
+    /// get storage cache manager
+    //  we need to disable the clippy warning here, lifetime eliding leads to compile errors
+    #[allow(clippy::needless_lifetimes)]
```

Comment:
I'm still a little confused... it seems strange to have a lifetime identifier here...

Reply:
Thanks for pointing this out!

After staring at these lines for quite a while (at least half a min, swear to the lamp :), finally recognized that its signature DOES NOT meet the lifetime elision rule.

> pub fn get_storage_cache_manager<'a>(self: &'a **Arc<Self>**) -> &'a CacheManager {
>       .....
> }

The type of `self` should have been a simple `&self` instead. really dumb...